x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -314,6 +330,37 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
         public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
             throw new IllegalArgumentException("[semantic_text] fields do not support sorting, scripting or aggregating");
         }
+
+        @Override
+        public Query createKnnQuery(byte[] queryVector, int numCands, Query filter, Float similarityThreshold, BitSetProducer parentFilter) {
+            return null;
+        }
+
+        @Override
+        public Query createKnnQuery(VectorData queryVector, int numCands, Query filter, Float similarityThreshold, BitSetProducer parentFilter, SearchExecutionContext context) {
+            if (parentFilter == null) {
+                Query parentFilterQuery = new FieldExistsQuery(SeqNoFieldMapper.PRIMARY_TERM_NAME);
+                parentFilter = context.bitsetFilter(parentFilterQuery);
+                // TODO Adjust for filterQuery
+            }
+
+            Query knnQuery = ((DenseVectorFieldMapper.DenseVectorFieldType) getEmbeddingsField().fieldType()).createKnnQuery(
+                queryVector,
+                numCands,
+                filter,
+                similarityThreshold,
+                parentFilter,
+                context
+            );
+
+            return new ESToParentBlockJoinQuery(knnQuery, parentFilter, ScoreMode.Avg, null);
```

Comment:
knn query implementation that delegates on the embeddings field, and wraps in a `ESToParentBlockJoinQuery`. Used to directly support knn queries to the field without using `nested` queries.